doc/developers/maintainer.rst.template

```diff
@@ -0,0 +1,356 @@
+      the following release checklist to the description of this PR to track the
+      progress.
+
+      .. code-block:: markdown
+
+        * [ ] Update news and what's new date in release branch
+        * [ ] Update news and what's new date and sklearn dev0 version in main branch
+        * [ ] Check that the wheels for the release can be built successfully
+        * [ ] Merge the PR with `[cd build]` commit message to upload wheels to the staging repo
+        * [ ] Upload the wheels and source tarball to https://test.pypi.org
+        * [ ] Create tag on the main repo
+        * [ ] Confirm bot detected at https://github.com/conda-forge/scikit-learn-feedstock
+              and wait for merge
+        * [ ] Upload the wheels and source tarball to PyPI
+        {%- if key != "rc" %}
+        * [ ] Publish to https://github.com/scikit-learn/scikit-learn/releases
+        {%- endif %}
+        * [ ] Announce on mailing list and on Twitter, and LinkedIn
+        {%- if key == "major" %}
+        * [ ] Update symlink for stable in https://github.com/scikit-learn/scikit-learn.github.io
+        {%- endif %}
+        {%- if key != "rc" %}
+        * [ ] Update SECURITY.md in main branch
+        {%- endif %}
+    
+    {% if key == "minor" %}
+    - Rebase this PR from the `{{ version }}.X` branch:
+
+      .. prompt:: bash
+
+        git rebase -i upstream/{{ version }}.X
+
+      This will open an interactive rebase with the `git-rebase-todo` containing all the
+      latest commits on `main`. At this stage, you have to perform this interactive
+      rebase with at least someone else (to not forget something and to avoid doubts).
+
+      - Do not remove lines but drop commit by replacing `pick` with `drop`.
+      - Commits to pick for a minor (bug-fix) release are *generally* prefixed with
+        `FIX`, `CI`, and `DOC`. They should at least include all the commits of the
+        merged PRs that were milestoned for this release and/or documented as such in
+        the changelog. It is likely that some bug fixes were documented in the changelog
+        of the next major release, in which case, the matching changelog entries will
+        need to be moved, first in `main` then backported in this release PR.
+      - Commits to `drop` for a minor (bug-fix) release are *generally* prefixed with
+        `FEAT`, `MAINT`, `ENH`, and `API`. Reasons for not including them is to prevent
+        change of behavior (which should only happen in major releases).
+      - After having dropped or picked commits, **do not exit** but paste the content of
+        the `git-rebase-todo` message in the PR. This file is located at
+        `.git/rebase-merge/git-rebase-todo`.
+      - Save and exit to start the interactive rebase. Resolve merge conflicts when
+        necessary.
+    {% endif %}
+
+    - Change the version numbers `__version__` in `sklearn/__init__.py` and `version` in
+      `pyproject.toml` to `{{ release }}`.
+
```

Comment:
Note that this section may have to be updated if we merge https://github.com/scikit-learn/scikit-learn/issues/29394